modules/core/src/main/java/com/google/refine/expr/Evaluable.java

```diff
@@ -66,4 +67,17 @@ public default Optional<Set<String>> getColumnDependencies(Optional<String> base
         return Optional.empty();
     }
 
+    /**
+     * Translates this expression by simultaneously substituting column names, as specified by the supplied map.
+     *
+     * @param substitutions
+     *            a map specifying new names for some columns. Keys of the map are old column names, values are the new
+     *            names for those columns. If a column name is not present in the map, the column is not renamed.
+     * @return a new expression with updated column names. If this renaming isn't supported, {@link Optional#empty()}
+     *         should be returned.
+     */
+    public default Optional<Evaluable> renameColumnDependencies(Map<String, String> substitutions) {
```

Comment:
After solving #133 based on this PR, I realized that the current approach of either renaming fully an operation or bailing out by returning `Optional.none()` isn't ideal. It feels more useful to me to have this function work on a best-effort basis, renaming as many column references as possible, but always producing an output.

In the case of renaming facets, you generally want to keep all your facets and have them renamed as fully as possible. Any remaining references to the former column should show up as facets with evaluation errors, which can then be fixed or deleted manually.

So I have a follow-up commit which changes that (changing the return type to `Evaluable` instead of `Optional<Evaluable>`), which has the benefit of simplifying the logic in a lot of implementations of this method. It depends on #6929 so I'll add this commit here once that one is merged.